.github/workflows/ci.yaml

```diff
@@ -22,9 +22,9 @@ jobs:
         run: |
           set -e
           sudo apt update
-          sudo apt -y install jo tox
-      - name: Test formatting with ruff
-        run: tox -e format -- --check
```

Comment:
Why not both? I mean, people do weird things on their computers all the time, I would not rely on local tests.

Otherwise an annoyed engineer that decides to bypass the pre-commit can poison the repo for everyone else.

Reply:
check the lines below, they run the pre-commit script, which executes `tox -e format -- --check`

Reply:
I should not review PRs when I'm tired. Sorry, of course.